internal/backend/service/register.go

```diff
@@ -61,9 +51,9 @@ func (s *RegisterServer) Register(
 		slogger = slog.Default()
 	}
 
-	serverID, err := getServerIDFromHeader(req)
+	serverID, err := interceptor.GetServerID(ctx)
```

Comment:
i don't understand how this is retrieving the server ID. what method is this calling?